### PR TITLE
feat(files): migrate template picker

### PR DESCRIPTION
**Pull request by skjnldsv**
Ref: https://github.com/nextcloud/server/issues/39914

image

### CODE REVIEW

**Comment by juliushaertl** [comment]
- Should also trigger the inline name editing after the file was created?
- I feel like we should still open the file directly after creating (as it used to be before) but not sure how easy that is to trigger after entering the name

**Comment by skjnldsv** [comment]
> * Should also trigger the inline name editing after the file was created?

Absolutely!

> I feel like we should still open the file directly after creating (as it used to be before) but not sure how easy that is to trigger after entering the name

Good point too! Can I do a followup? :)

**Comment by skjnldsv** [comment]
Ping everyone!

**Review by juliusknorr — APPROVED** [review]

File: apps/files/src/views/TemplatePicker.vue [hunk 1]
```diff
@@ -199,12 +209,11 @@ export default {
 
 		async onSubmit() {
 			this.loading = true
-			const currentDirectory = getCurrentDirectory()
-			const fileList = OCA?.Files?.App?.currentFileList
+			const currentDirectory = new URL(window.location.href).searchParams.get('dir') || '/'
```
Comment (artonge) [inline apps/files/src/views/TemplatePicker.vue:212]: Why can't we use something like ` this.$route.params.dir` ?

File: apps/files/src/views/TemplatePicker.vue [hunk 2]
```diff
@@ -216,35 +225,43 @@ export default {
...
-						fileList,
-						fileActions: fileList?.fileActions,
-						fileInfoModel: model,
-					})
-				}
+				// Update files list
+				emit('files:node:created', node)
 
+				// Open the new file
+				window.OCP.Files.Router.goToRoute(
```
Comment (artonge) [inline apps/files/src/views/TemplatePicker.vue:248]: Wasn't there another way to navigate without using `window.OCP` ?

**Review by artonge — APPROVED** [review]

File: apps/files/src/views/TemplatePicker.vue (hunk 1 above)
Comment (skjnldsv) [reply apps/files/src/views/TemplatePicker.vue:212]: The templatepicker is standalone and not aware of the Files app router.
We do not expose the route raw to keep things clean and limit what devs can do with it

File: apps/files/src/views/TemplatePicker.vue (hunk 2 above)
Comment (skjnldsv) [reply apps/files/src/views/TemplatePicker.vue:248]: Same as above: The templatepicker is standalone` and works without the Files app

**Pull request merged**